dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserSettingService.java

```diff
@@ -177,14 +178,14 @@ public Serializable getUserSetting(UserSettingKey key, String username) {
   @Transactional(readOnly = true)
   public Map<String, Serializable> getUserSettingsWithFallbackByUserAsMap(
       User user, Set<UserSettingKey> userSettingKeys, boolean useFallback) {
-    Map<String, Serializable> result =
-        Sets.newHashSet(getUserSettings(user)).stream()
-            .filter(
-                userSetting ->
-                    userSetting != null
-                        && userSetting.getName() != null
-                        && userSetting.getValue() != null)
-            .collect(Collectors.toMap(UserSetting::getName, UserSetting::getValue));
+    Map<String, Serializable> result = new HashMap<>();
+    Sets.newHashSet(getUserSettings(user)).stream()
```

Comment:
Should use `Set.of` instead of `Sets.newHashSet` but if the next step is to use `stream()` there is no need to "copy" the result of `getUserSettings(user)`, a `stream()` processing is not affecting the iterated collection.

Reply:
nice catch, this is an overlook when I fix the null issue of the code.